src/test/ui/consts/dangling-alloc-id-ice-2.rs

```diff
@@ -0,0 +1,10 @@
+// https://github.com/rust-lang/rust/issues/55223
+
+struct Slice(&'static [&'static [u8]]);
+
+static MAP: Slice = Slice(&[
+    b"CloseEvent" as &'static [u8],
+]);
```

Comment:
I don't even understand why anything dangles here...?

Reply:
it shouldn't, but it does. That's the regression. I think it's related to the explicit coercion cast

Reply:
But it's a regression not fixed by this PR, right? This test makes it look like what it tests for is intended behavior.

Reply:
No, this PR just fixes the ICE to be a validation error. I commented on the issue and pinged the appropriate people. I don't know how to fix this regression